### ISSUE TITLE
Find Transactions returns duplicate transaction hashes

### ISSUE THREAD

**Issue by DyrellC** [iotaledger/hornet#620]
**Describe the bug**
When using the find transactions request on my hornet node, I noticed that there were duplicate transaction hashes being returned. When looking into the `webapi/transactions/findTransactions` code I found that the list of return hashes is an array that does not remove duplicates, and appends transactions through the individual query fields rather than ensuring that the transaction referenced has all the fields individually accounted for. 

This means that if I have 2 transactions with the same address, (but different tags), and i make a `findTransactions` query with the address and tag fields included, it will return 3 transaction hashes, 2 of which are the same (1 from the addresses fetch and 1 from the tag fetch). I would assume that this is not the intended behavior, as it is specified in https://docs.iota.works/docs/iri/1.0/references/iri-api-reference#findtransactions that the fields should be returned on an AND basis and not an OR.   

**To Reproduce**
- Send two transactions to the same address but with different tags. 
- Call find transactions with the address and one of the tags specified
- Check the return 

**Expected behavior**
Only one of these hashes should be returned: The one that includes both the transaction hash as well as the tag, not a combination.

**Environment information:**
 - OS: Windows 10
 - RAM: 16GB
 - Cores: 8 Cores
 - Type: Desktop
 - HORNET version 0.4.2

**Additional context**
You can see the transactions here: 
https://thetangle.org/address/RGIIUERDEEMABCEDEGLEKDRGNEZCWDS9ODIHHGDEYBSCSBFFJAAEUAGBRAMDW9YCCCKBXBQBVBOBKBDC9OREANLKHW

Query with `Addresses: ['RGIIUERDEEMABCEDEGLEKDRGNEZCWDS9ODIHHGDEYBSCSBFFJAAEUAGBRAMDW9YCCCKBXBQBVBOBKBDC9OREANLKHW']`
```
hashes: [
    'ZEKHEPBJFKKBNEXOGNATGV9MNDUCEFEIKTODCLDK9FSUUMWJBGRYMMBDUZIQZFEVTLKIO9QEAUWK99999',
    'SQMIF9QLAZSLQVUEZLUMXCAGWCHBHGQWKVFVDTDDLDLKQQMTURWJS9EEBHNSTRSRFILNMNRJEFDK99999'
  ],
```

Query with `Tags: ['O9EIMBDA9HAIF9RAE9RAFDDG999']`
```
 hashes: [
    'SQMIF9QLAZSLQVUEZLUMXCAGWCHBHGQWKVFVDTDDLDLKQQMTURWJS9EEBHNSTRSRFILNMNRJEFDK99999'
  ],
```

Query with `Addresses: ['RGIIUERDEEMABCEDEGLEKDRGNEZCWDS9ODIHHGDEYBSCSBFFJAAEUAGBRAMDW9YCCCKBXBQBVBOBKBDC9OREANLKHW']`
 and `Tags: ['O9EIMBDA9HAIF9RAE9RAFDDG999']`
```
 hashes: [
    'ZEKHEPBJFKKBNEXOGNATGV9MNDUCEFEIKTODCLDK9FSUUMWJBGRYMMBDUZIQZFEVTLKIO9QEAUWK99999',
    'SQMIF9QLAZSLQVUEZLUMXCAGWCHBHGQWKVFVDTDDLDLKQQMTURWJS9EEBHNSTRSRFILNMNRJEFDK99999',
    'SQMIF9QLAZSLQVUEZLUMXCAGWCHBHGQWKVFVDTDDLDLKQQMTURWJS9EEBHNSTRSRFILNMNRJEFDK99999'
  ],
```

**Comment by muXxer** [comment]
Actually, the good ol' API is broken by design, that's why we implemented it as an OR instead of an AND.

For example, what should happen if someone passes 2 Tags? Should we only search for transactions that contain both tags at the same time? Or should we search all tx that have one of these tags? Sure, second option. But why does this behavior suddenly change if you pass tx hashes? Makes absolutely no sense at all, this whole call. I really would like to block the usage of different "filters" at the same time.

**Comment by kwek20** [comment]
@muXxer Well I do believe the idea was that the Hornet API was interchangeable with the old IRI API, which would mean the same output.
As nobody brought up this issue before, it doesn't seem to be used a lot, but its definitely breaking for us. Its forcing us to call `findTransactions` twice, once for tag and once for address, and then locally filter. This is clearly not efficient and not an option for small devices on the edge. 

I do see the issue with multiple tags, we could propose a default behaviour? Leaving an undefined/broken call is 100% worse than defining it :)

**Comment by muXxer** [comment]
My problem with this is, if you filter it in the node, you can't limit the amount of keys it will iterate, otherwise it will lead to wrong responses. Making it a really heavy operation. Maybe I'm wrong here but it could be used for DOS attacks

**Comment by muXxer** [comment]
Fixed in #637
